### PR TITLE
[B2BP-418] Added capability for internal icon use

### PR DESCRIPTION
**Pull request by simosalsi**
As per title.
This PR lays the foundation for the use of internal icons when MUI icons is not enough.

#### List of Changes

- Added logic for internal icons use
- Added `Medium` as a valid icon for Footer's Social Links
- Updated `Footer` to include the use of the new internal icon

-- Update --
- Subbed internal Medium icon for MUIItalia's MediumIcon (logic remains the same)

#### Motivation and Context

Not all icons needed for the replication of SEND's website are included in MUI icons.

#### How Has This Been Tested?

Tested locally.

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/nextjs-website/src/components/icons/Medium.tsx
Comment (kin0992) [inline apps/nextjs-website/src/components/icons/Medium.tsx:1]: I think you can use the [Medium Icon from `mui-italia`](https://main--633c31eff9fe385398ada426.chromatic.com/?path=/docs/assets-icons--docs)

**Review by kin0992 — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/components/Icon.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import {
+  Facebook,
+  HelpOutlineOutlined,
+  Instagram,
+  LinkedIn,
+  Twitter,
+} from '@mui/icons-material';
+import Medium from './icons/Medium';
```
Comment (datalek) [inline apps/nextjs-website/src/components/Icon.tsx:8]: Check out if it is the same icon:

```suggestion
import { MediumIcon } from '@pagopa/mui-italia/dist/icons/MediumIcon';
```

File: apps/nextjs-website/src/components/Icon.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+const iconMapping = {
+  // Button Icons
+  HelpOutlineOutlined: <HelpOutlineOutlined />,
+
+  // Social Icons
+  Instagram: <Instagram />,
+  LinkedIn: <LinkedIn />,
+  Facebook: <Facebook />,
+  Twitter: <Twitter />,
+  Medium: <Medium />,
```
Comment (datalek) [inline apps/nextjs-website/src/components/Icon.tsx:19]:
```suggestion
  Medium: <MediumIcon />,
```

File: apps/nextjs-website/src/components/icons/Medium.tsx
Comment (datalek) [inline apps/nextjs-website/src/components/icons/Medium.tsx:1]: If `@pagopa/mui-italia/dist/icons/MediumIcon` is the same icon you can remove this file

**Review by datalek — APPROVED** [review]

**Review by kin0992 — APPROVED** [review]

**Pull request merged**